### PR TITLE
TMA-366 Add logs subscription for CSLS for all TxMA log groups

### PR DESCRIPTION
**Pull request by jashelb90**
We need to ship our logs to CSLS so they can be monitored by the cyber team.

The following PR contains new resources in CloudFormation to create a CloudWatch Logs Subscription Filter per Log Group owned by TxMA.

I have also removed an unused policy statement.

*In Progress: S3 logs shipping

### CODE REVIEW

File: event-processing/event-processing-template.yml [hunk 1]
```diff
@@ -2024,3 +2047,124 @@ Resources:
...
+      Bucket: !Ref FraudSplunkDeliveryFailureBucket
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: 'Allow'
+            Resource:
+              - !GetAtt FraudSplunkDeliveryFailureBucket.Arn
+              - !Sub '${FraudSplunkDeliveryFailureBucket.Arn}/*'
+            Principal:
+              AWS: "{{resolve:ssm:CSLSS3LambdaARN}}"
```
Comment (denizgenc) [inline event-processing/event-processing-template.yml:2149]: I don't know what's in this SSM parameter, but to make sure: it should be the Lambda role ARN, not the ARN for the lambda itself.

File: audit/audit-template.yml [hunk 2]
```diff
@@ -178,6 +179,18 @@ Resources:
...
+        !If
+          - IsProduction
+          -
+            QueueConfigurations:
+              - Event: "s3:ObjectCreated:*"
+                Queue: "{{resolve:ssm:CSLSS3QueueARN}}"
+              - Event: "s3:ObjectRemoved:*"
+                Queue: "{{resolve:ssm:CSLSS3QueueARN}}"
+              - Event: "s3:ObjectRestore:*"
+                Queue: "{{resolve:ssm:CSLSS3QueueARN}}"
```
Comment (denizgenc) [inline audit/audit-template.yml:192]: AFAIK we have only ever dealt with `s3:ObjectCreated` notifications. `s3:ObjectRestore` might work. Notifying on `s3:ObjectRemoved` doesn't seem like a good idea though, since the S3 processor tries to retrieve the S3 object referred to in the notification. Looking at the CSLS source code:
```python
            for record in lambda_event["Records"]:
                sqs_body = json.loads(record["body"])
                for message in sqs_body["Records"]:
                    ...
                    s3_object = s3_get_object(bucket, key)
                    if s3_object:
                        ...
                    else:
                        LOG.error("Unable to retrieve S3 object")
````
`s3_get_object()` wraps a `GetObject` request in a try/except statement, so the whole lambda shouldn't fall over if the object doesn't exist (which would be the case for `s3:ObjectRemoved` notifications, I assume). However I think it would be better for various reasons (such as performance of the lambda) if it wasn't notified for events that no longer exist. If you want to delete events in Splunk itself, I'm afraid CSLS can't do that.

tl;dr: You shouldn't notify on `s3:ObjectRemoved`

File: event-processing/event-processing-template.yml [hunk 3]
```diff
@@ -1204,6 +1216,17 @@ Resources:
...
+      NotificationConfiguration:
+        !If
+        - IsProduction
+        - QueueConfigurations:
+            - Event: "s3:ObjectCreated:*"
+              Queue: "{{resolve:ssm:CSLSS3QueueARN}}"
+            - Event: "s3:ObjectRemoved:*"
+              Queue: "{{resolve:ssm:CSLSS3QueueARN}}"
+            - Event: "s3:ObjectRestore:*"
+              Queue: "{{resolve:ssm:CSLSS3QueueARN}}"
```
Comment (denizgenc) [inline event-processing/event-processing-template.yml:1228]: See earlier comment

File: event-processing/event-processing-template.yml [hunk 4]
```diff
@@ -1126,6 +1127,17 @@ Resources:
...
```
Comment (denizgenc) [inline event-processing/event-processing-template.yml:1139]: See earlier comment

File: audit/audit-template.yml [hunk 5]
```diff
@@ -337,24 +350,33 @@ Resources:
...
+      Bucket: !Ref MessageBatchBucket
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: 'Allow'
+            Resource:
+              - !GetAtt MessageBatchBucket.Arn
+              - !Sub '${MessageBatchBucket.Arn}/*'
+            Principal:
+              AWS: "{{resolve:ssm:CSLSS3LambdaARN}}"
```
Comment (denizgenc) [inline audit/audit-template.yml:379]: Reminder again to ensure that this ARN is the Lambda role arn

**Review by denizgenc — CHANGES_REQUESTED** [review]
I don't think we should notify on `s3:ObjectRemoved`, and would like to make sure that the `{{resolve:ssm:CSLSS3LambdaARN}}` resolves to the Lambda _role_ ARN.

File: event-processing/event-processing-template.yml (hunk 1 above)
Comment (jashelb90) [reply event-processing/event-processing-template.yml:2149]: Yeah, that's correct. I have just shortened the parameter name

**Review by denizgenc — APPROVED** [review]
Looks all good from here!

**Review by Aidan23Smith — APPROVED** [review]
Looks good, and needs to be tested

**Review by Aidan23Smith — APPROVED** [review]
Still the same
